cli/root.go

```diff
@@ -85,15 +83,30 @@ func NewAlertmanagerClient(amURL *url.URL) *client.Alertmanager {
 
 	cr := clientruntime.New(address, path.Join(amURL.Path, defaultAmApiv2path), schemes)
 
-	cr.Transport = &http.Transport{
-		TLSClientConfig: tlsConfig,
+	if amURL.User != nil && httpConfigFile != "" {
+		kingpin.Fatalf("basic authentication and http.config.file are mutually exclusive")
```

Comment:
I'm wondering what takes precedence if you specify credentials both in the URL and in the HTTP config file?

Reply:
That is mutually exclusive. You cannot specify credentials both in the URL and in the HTTP config file. If you specify credentials both in the URL and in the HTTP config file, the error `basic authentication and http.config.file are mutually exclusive` thrown and exit.

Reply:
I get that but my question was more: how would `net/http` behave if we don't forbid this combination? I've tested locally and the credentials from the config file take precedence over the credentials passed in the URL. I'm fine with keeping this check though since it would avoid people losing time figuring out why their credentials aren't taken into account.

Reply:
Yes. keep this check make it clear to avoid such thing you mentioned. Thanks @simonpasquier